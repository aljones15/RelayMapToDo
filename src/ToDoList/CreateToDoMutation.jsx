import { commitMutation, graphql } from 'react-relay';
import environment from '../../data/relayEnv';

const mutation = graphql`
  mutation CreateToDoMutation($input: ToDoInput!)
  {
    createToDo(input: $input){
      ...ToDo      
    }
  }
`;

export default function AddToDo(city_id, text) {
  const variables = {
    input: {
      city_id: city_id,
      text: text
    }
  };
  commitMutation(
    environment,
    {
      mutation,
      variables,
      onCompleted: (response) => {},
      onError: err => console.error(err)
    }
  );
}
