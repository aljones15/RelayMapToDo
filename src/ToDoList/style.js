export const ToDoStyle = {
  display: 'flex',
  flexGrow: 1,
  flexShrink: 1,
  flexDirection: 'column',
  padding: '0px',
};

export const ToDoAddStyle = {
  display: 'flex',
  flexGrow: 1,
  flexShrink: 1,
  color: '#FFFFFF',
};
