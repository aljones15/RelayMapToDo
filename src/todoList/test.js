import React from 'react';
import renderer from 'react-test-renderer';
import ToDoList from './index';

describe('ToDoList Should', () => {
  const elJson = renderer.create(<ToDoList />).toJSON();
  it('match a snapshot', () => {
    expect(elJson).toMatchSnapshot();
  });
});
