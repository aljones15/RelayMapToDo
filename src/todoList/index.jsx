import React from 'react';
import ToDo from '../todo/index.jsx';
import {ToDoStyle, ToDoAddStyle} from './style';

const ToDoList = (props) => {
  return( 
    <ul className='ToDoUl' style={ToDoStyle}>
    <li style={ToDoAddStyle} >Know Something to do here?</li>
    {	  
      props.list.map((todo, index) => <ToDo key={index + '_todo'}  />)
    }
    </ul>
  )
}

export default ToDoList
