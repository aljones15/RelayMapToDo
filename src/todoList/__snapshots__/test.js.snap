// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Should match a snapshot 1`] = `
<ul>
  To Do!
</ul>
`;
